packages/core/src/mcp/index.ts

```diff
@@ -84,10 +86,12 @@ export abstract class MCPServerBase extends MastraBase {
 
   /**
    * Abstract method to convert and validate tool definitions provided to the server.
+   * This method will also handle agents passed in the config.
    * @param tools Tool definitions to convert.
+   * @param agents Agent definitions to convert to tools.
    * @returns A record of converted and validated tools.
    */
-  public abstract convertTools(tools: ToolsInput): Record<string, ConvertedTool>;
+  public abstract convertTools(tools: ToolsInput, agents?: MCPServerConfig['agents']): Record<string, ConvertedTool>;
```

Comment:
I don't know if I agree with this TBH.

Why aren't we letting the user do the work?
```
{
  agents: [agentToTool(agent, generatoptions?) // maybe you want to set temperature etc so you can model things here
}

Reply:
I was thinking that eventually, `generateOptions` should be part of the tool call inputSchema. So the tool call can dynamically populate the generateOptions. But for now I was just doing the simplest thing possible so we can see how people use this feature and what people ask for with it.

Reply:
The whole purpose of this is to make it as automatic and frictionless as possible